File: School.Domain/Entities/Class.cs
using System;
using System.Collections.Generic;

namespace School.Domain.Entities
{
    public class Class
    {
        public int Id { get; set; }
        public int TeacherId { get; set; }
        public Teacher Teacher { get; set; }
        public DateTime Start { get; set; }
        public DateTime End { get; set; }
        public int UnitId { get; set; }
        public Unit Unit { get; set; }
        public int roomId { get; set; }
        public Room Room { get; set; }
    }
}

File: Views/src/app/organization/interfaces/Class.ts
import { Teacher } from './Teacher';
import { Discipline } from './Discipline';
import { Student } from './Student';
import { Unit } from './Unit';
import { Room } from './Room';

export class Class {

    id: number;
    teacherid: number;
    teacher: Teacher;
    start: Date;
    end: Date;
    disciplineid: number;
    discipline: Discipline;
    studentid: number;
    students: Student[];
    unitid: number;
    unit: Unit;
    roomid: number;
    room: Room;

}

